### DISCUSSION TITLE
Cryptomator Issues

### DISCUSSION THREAD

**Discussion by xclodude** [iterate-ch/cyberduck#15029] [Q&A] [answered]
I'm hoping someone can help me with my Cryptomator problems whiles using Mountainduck with Storj.
Here are the steps I tried and the issues that occurred. 
Step 1: I mounted a drive to Storj, appears no issues. Created a folder, uploaded couple documents un-encrypted, no issues. 
Step 2: I attempted to create a cryptomator vault within windows explorer by right clicking a folder under my cloud drive > Mountain Duck > Cryptomator > New Encrypted Vault
It prompted to name the vault, create a passphrase, etc. I went through that process and to the end. ISSUE: It only created a folder, but no vault, and no vault with a drive letter appeared within explorer
I tried this with just Mountain Duck installed and with both the Mountain Duck and separate Cryptomator apps installed. It acted the same both times. 
Step 3: I then attempted to create a vault using the Cryptomator program. I went through the Cryptomator steps with no issues, it created a vault folder within my cloud folder, and appeared to create the vault. The first odd thing was that both Cryptomator AND Mountain Duck asked for the passphrase which seemed odd. At first everything seemed fine. Cryptomator showed the vault, explorer showed the vault, etc. ISSUE: Then all of a sudden while trying to use the vault, windows explorer indicated that the pathway no longer existed, and Cryptomator seemed to error. If I locked the vault, it then showed "Cryptomator could not find a vault at this path" (to note: I was still mounted to storj, that hadn't changed and was showing good connection). The biggest oddity is that the Mountain Duck Sync activity indicated that multiple files and folders were deleted, however I didn't delete them. 
Step 4: I turned off "auto detect and open vault in browser" and again created a test vault within the mounted cloud drive. This appeared to mostly work, and I was able to upload a 100mb file into the vault, lock, unlock, and it seemed no major issues. ISSUE: The only odd thing is that folder "c" within the non-encrypted portion of the vault location appeared to again "delete" on its own. From the reading I did, I believe Cryptomator needs that "c" folder. I tried remaking one, and it again deleted. Maybe this is supposed to happen.
In the end though, I would like to use Mountain Duck with the baked in Cryptomator and that doesn't seem to work. 
Step 5: I re-checked "Auto detect and open vault in browser" and multiple Cryptomator files and folders deleted and again Cryptomator could not find a vault at the path.

**Comment by tatze96** [comment, marked as answer]
Regarding your mentioned issues:
> It only created a folder, but no vault, and no vault with a drive letter appeared within explorer
I tried this with just Mountain Duck installed and with both the Mountain Duck and separate Cryptomator apps installed. It acted the same both times.

Cryptomator is fully integrated into Mountain Duck and Cyberduck, so you generally don't have to have the Cryptomator application installed separately.

The vault is expected not to be mounted with a separate drive letter as it gets decrypted in place within the mounted drive. If you don't see a "d" folder, a "masterkey.cryptomator" file, and a "vault.cryptomator" file within the created vault, you most likely have the [automatic vault detection](https://docs.cyberduck.io/cryptomator/#auto-detect) activated. The automatic vault detection causes Mountain Duck to automatically decrypt the vault given that the password is saved within the keychain. The vault then appears to behave as a regular folder, although it is a Cryptomator vault.

>Then all of a sudden while trying to use the vault, windows explorer indicated that the pathway no longer existed, and Cryptomator seemed to error. If I locked the vault, it then showed "Cryptomator could not find a vault at this path" (to note: I was still mounted to storj, that hadn't changed and was showing good connection). The biggest oddity is that the Mountain Duck Sync activity indicated that multiple files and folders were deleted, however I didn't delete them.

This is expected as the Cryptomator application can't see the necessary "masterkey.cryptomator" file anymore if Mountain Duck has decrypted the vault. Once Mountain Duck has decrypted the vault, the previously mentioned "d" folder, "masterkey.cryptomator" file, and "vault.cryptomator" file gets removed from the local file listing and replaced by the vault content until the vault gets locked again. This also causes the delete entries within the sync activity.

> The only odd thing is that folder "c" within the non-encrypted portion of the vault location appeared to again "delete" on its own. From the reading I did, I believe Cryptomator needs that "c" folder. I tried remaking one, and it again deleted. Maybe this is supposed to happen.

I'm not sure to which folder you are referring. Please send us screenshots or a screencast to visualize the issue. Make sure to attach it through the GitHub web interface, as it wouldn't be visible to us otherwise.
